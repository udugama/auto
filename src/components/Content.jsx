import React from 'react';
import { Paper } from '@material-ui/core';
import Story from './Story';

export default class Content extends React.Component {

    render() {
        const listData = this.props.content;
        let storiesList = 'Stories List';
        if (listData) {
            storiesList = listData.map((row) => {
                return (
                    <Story key={row.objectID} data={row} />
                );
            });
        }
        return (
            <Paper style={{maxHeight: 400, overflow: 'auto'}} >
                {storiesList}
            </Paper>
        );
    }
}