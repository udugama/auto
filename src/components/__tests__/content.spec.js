import React from 'react';
import ReactDOM from 'react-dom';
import Content from './../Content';
import { shallow } from 'enzyme';
import { data } from './../fixtures/data';
import { render } from '@testing-library/react';

const { hits } = data;

it('renders without crashing', () => {
    shallow(<Content content={hits} />);
});

it('render set of stories based on data array', () => {
  const { getByText } = render(<Content content={hits} />);
  expect(getByText).toMatchSnapshot();
});