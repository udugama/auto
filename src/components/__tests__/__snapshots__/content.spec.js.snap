// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render set of stories based on data array 1`] = `[Function]`;
